repair/featurize/occurfeat.py

```diff
@@ -55,28 +74,53 @@ def create_tensor(self):
         return combined
 
     def gen_feat_tensor(self, row, tuple):
+        """
+        For a given cell, we calculate the co-occurence probability of all domain values
+        for row['attribute'] with the row's co-value in every co-attributes.
+
+        That is for a domain value 'd' and current co-attribute value 'c' we have
+            P(d | c)   P(d, c) / P(c)
+
+        where P(d,c) is the empirical co-occurrence frequency of 'd' and 'c' and P(c)
+        is the frequency of 'c'.
+        """
+        # tensor is a (1 X domain size X # of attributes) pytorch.Tensor
+        # tensor[0][domain_idx][coattr_idx] contains the co-occurrence probability
+        # between the co-attribute (coattr_idx) and the domain values
+        # a possible domain value for this entity (row/tuple)
         tensor = torch.zeros(1, self.classes, self.attrs_number)
         rv_attr = row['attribute']
+        # Domain value --> index mapping
         domain = row['domain'].split('|||')
         rv_domain_idx = {val: idx for idx, val in enumerate(domain)}
+
+        # Iterate through every attribute (and current value for that
+        # attribute) and set the co-occurrence probability for every
+        # domain value for our current row['attribute'].
         for attr in self.all_attrs:
-            if attr != rv_attr and (not pd.isnull(tuple[attr])):
```

Comment:
This is mostly a style change where I changed this if statement to a guard block below.